src/tools/openpgp-tool.c

```diff
@@ -419,7 +435,7 @@ static int do_dump_do(sc_card_t *card, unsigned int tag)
 		if (sizeof(buffer) != r)
 			return EXIT_FAILURE;
 	} else {
-		util_hex_dump_asc(stdout, buffer, sizeof(buffer), -1);
+		util_hex_dump_asc(stdout, buffer, r, -1);
```

Comment:
I think `r` also needs to be respected in line 410:
```
		if (fp) {
			r = (int)fwrite(buffer, sizeof(char), sizeof(buffer), fp);
		}
```
... and line 419-420 needs to be removed:
```
		if (sizeof(buffer) != r)
			return EXIT_FAILURE;
```
or am I getting this wrong?